File: extras/binary_search.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int n = 10;
    int array[] = {1,2,3,4,5,5,3,5,2,122};
    int a = 0, b = n-1;
    int x = 10;
    while (a <= b) {

        int k = (a+b)/2;

        if (array[k] == x) {
            // x found at index k
        }

        if (array[k] > x) b = k-1;

        else a = k+1;
        }
}

File: extras/binary_search.py
list  = [1, 2, 4, 6, 10, 11, 15, 16, 20, 22, 26]


search = 1

a = 0
b = len(list) - 1

while (a <= b):
    k = int((a+b) / 2)

    if (list[k] == search):
        print(f"found, at index: {k}")
        break

    if(list[k] < search):
        a = k + 1

    else:
        b = k - 1